bazelisk.py

```diff
@@ -92,26 +94,36 @@ def resolve_version_label_to_number(bazelisk_directory, version):
 
 
 def determine_bazel_filename(version):
-  machine = platform.machine()
+  machine = normalized_machine_arch_name()
   if machine != 'x86_64':
     raise Exception('Unsupported machine architecture "{}". '
                     'Bazel currently only supports x86_64.'.format(machine))
 
-  operating_system = platform.system().lower()
-  if operating_system not in ('linux', 'darwin', 'windows'):
+  if OPERATING_SYSTEM not in ('linux', 'darwin', 'windows'):
     raise Exception('Unsupported operating system "{}". '
                     'Bazel currently only supports Linux, macOS and Windows.'
-                    .format(operating_system))
+                    .format(OPERATING_SYSTEM))
+
+  if machine == 'amd64':
+    machine = 'x86_64'
+  return 'bazel-{}-{}-{}'.format(version, OPERATING_SYSTEM, machine)
 
-  return 'bazel-{}-{}-{}'.format(version, operating_system, machine)
+
+def normalized_machine_arch_name():
+  machine = platform.machine().lower()
+  if machine == 'amd64':
+    machine = 'x86_64'
+  return machine
 
 
 def determine_url(version, bazel_filename):
   # Split version into base version and optional additional identifier.
   # Example: '0.19.1' -> ('0.19.1', None), '0.20.0rc1' -> ('0.20.0', 'rc1')
   (version, rc) = re.match(r'(\d*\.\d*(?:\.\d*)?)(rc\d)?', version).groups()
+
+  url_ending = '.exe' if OPERATING_SYSTEM == 'windows' else ''
```

Comment:
I think it'd be easier to understand the flow of the data if `determine_bazel_filename` simply returned the correct filename with the .exe suffix, instead of adding it here. Otherwise wouldn't we also save it locally to a file not with the .exe suffix and how can we then even execute it on Windows?

Reply:
> Otherwise wouldn't we also save it locally to a file not with the .exe suffix and how can we then even execute it on Windows?

Ah, good question. I've just realised that Bazelisk saves Bazel as `~/.bazelisk/bin/bazel-0.20.0-windows-x86_64`, without a `.exe`, and yet it still works (which I assume is because Bazelisk still reads and runs it as if it were a Windows executable).

I agree that it'd be easier to understand if `determine_bazel_filename` returned the correct filename itself rather than delegating it to `determine_url`, so I'll make that change. :+1:

Reply:
Done.